operator/identity/gc_test.go

```diff
@@ -0,0 +1,298 @@
+						},
+					},
+				),
+			},
+		)
+		if err == nil && len(identities.Items) == 1 {
+			break
+		}
+		time.Sleep(50 * time.Millisecond)
+	}
+	if err != nil {
+		t.Fatalf("unable to list Cilium identities: %s", err)
+	}
+	if len(identities.Items) != 1 {
+		t.Fatalf("expected 1 Cilium identity, got %d", len(identities.Items))
+	}
+	if identities.Items[0].Name != "99999" {
+		t.Fatalf("expected Cilium identity \"99999\", got %q", identities.Items[0].Name)
+	}
+
+	if err := hive.Stop(ctx); err != nil {
+		t.Fatalf("failed to stop: %s", err)
+	}
+}
+
+func setupK8sNodes(clientset k8sClient.Clientset) error {
+	nodes := []*corev1.Node{
+		{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "v1",
+				Kind:       "Node",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   "node-control-plane",
+				Labels: map[string]string{"kubernetes.io/hostname": "node-control-plane"},
+			},
+		},
+		{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "v1",
+				Kind:       "Node",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   "node-worker",
+				Labels: map[string]string{"kubernetes.io/hostname": "node-worker"},
+			},
+		},
+	}
+	for _, node := range nodes {
+		if _, err := clientset.CoreV1().Nodes().
+			Create(context.Background(), node, metav1.CreateOptions{}); err != nil {
+			return fmt.Errorf("failed to create node %v: %w", node, err)
+		}
+	}
+	return nil
+}
+
+func setupCiliumIdentities(clientset k8sClient.Clientset) error {
+	identities := []*v2.CiliumIdentity{
+		{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "cilium.io/v2",
+				Kind:       "CiliumIdentity",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "88888",
+				Labels: map[string]string{
+					"test": "identities-gc",
+				},
+			},
+		},
+		{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "cilium.io/v2",
+				Kind:       "CiliumIdentity",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "99999",
+				Labels: map[string]string{
+					"test": "identities-gc",
+				},
+			},
+		},
+	}
+	for _, identity := range identities {
+		if _, err := clientset.CiliumV2().CiliumIdentities().
+			Create(context.Background(), identity, metav1.CreateOptions{}); err != nil {
+			return fmt.Errorf("failed to create identity %v: %w", identity, err)
+		}
+	}
+	return nil
+}
+
+func setupCiliumEndpoint(clientset k8sClient.Clientset) error {
+	endpoint := &v2.CiliumEndpoint{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-endpoint",
+		},
+		Status: v2.EndpointStatus{
+			Identity: &v2.EndpointIdentity{
+				ID: 99999,
+			},
+		},
+	}
+	if _, err := clientset.CiliumV2().CiliumEndpoints("").
+		Create(context.Background(), endpoint, metav1.CreateOptions{}); err != nil {
+		return fmt.Errorf("failed to create endpoint %v: %w", endpoint, err)
+	}
+	return nil
+}
+
+func registerIdentitiesGC(
+	lc hive.Lifecycle,
+	params gcParams,
+	cfg GCConfig,
+	sharedCfg GCSharedConfig,
+) {
+	// create the gc (it will be managed through hive lifecycle hooks)
+	_ = newGC(params, cfg, sharedCfg)
```

Comment:
I'd not bury this in here. You can just do `cell.Invoke(newGC)` inside `hive.New` above. And rename this method to `setupCiliumEndpointWatcher` or something.